tests/PHPStan/Analyser/data/non-empty-string-substr-specifying.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace NonEmptyStringSubstr;
+
+use function PHPStan\Testing\assertType;
+
+class Foo {
+	public function nonEmptySubstr(string $s, int $offset, int $length): void
+	{
+		if (substr($s, 10) === 'hallo') {
+			assertType('non-empty-string', $s);
+		}
+		assertType('string', $s);
+
+		if (substr($s, -10) === 'hallo') {
+			assertType('non-empty-string', $s);
+		}
+		assertType('string', $s);
+
+		if (substr($s, 10, 5) === 'hallo') {
+			assertType('non-empty-string', $s);
+		}
+		assertType('string', $s);
+
+		if (substr($s, 10, -5) === 'hallo') {
+			assertType('non-empty-string', $s);
+		}
+		assertType('string', $s);
+
+		if (substr($s, $offset) === 'hallo') {
+			assertType('non-empty-string', $s);
+		}
+		assertType('string', $s);
+
+		if (substr($s, $offset, $length) === 'hallo') {
```

Comment:
There should be a test for yoda condition (reversed) too.

Reply:
done 👍